MC-35509: Updated the software update guide module with the safe upgrade tool documentation (#8314)
magento/devdocs: src/safe-upgrade-tool/install.md

---
group: software-update-guide
title: Install
ee_only: True
functional_areas:
  - Upgrade
---

The Safe Upgrade Tool ALPHA (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed on it.

{:.bs-callout-warning}
At the moment this is an ALPHA version with limited scope, only validating PHP Magento APIs and GraphQL schema.

## Workflow

The following diagram shows the expected workflow when running the SUT:

> ![SUT Diagram](img/mvp-diagram.png)

Reviewer: Please replace this image with this one:
![image](https://user-images.githubusercontent.com/10559423/105175070-bf83c900-5b23-11eb-9a23-a139e53211e8.png)

Author: added

### Who is the SUT for?

> 

Reviewer: Let's add this here:
"The SUT Alpha is now available for all Magento Commerce merchants, and it can be downloaded on Magento repository [repo link]."

### SUT main use case

Author: added link

The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:

> 

Reviewer: Let's add this image here
![image](https://user-images.githubusercontent.com/10559423/105175833-bba47680-5b24-11eb-87b5-bd350f8efa41.png)

Author: added

Reviewer: I can't see it but great that you add it. Thanks

1. The Account Manager creates a timeline and cost for the Magento upgrade, which allows them to get their manager's approval.
1. With their manager's approval, the Software Engineer works on the required code modifications to fix the broken modules.
1. The Software Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.
1. Everything check’s out and the Software Engineer pushes the code to a staging environment where regression tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.

## Prerequisites

See [prerequisites]({{site.baseurl}}/safe-upgrade-tool/prerequisites.html) for more information.

{:.bs-callout-info}
You can run the SUT in any operating system. There is no requirement to run the SUT where your Magento instance is located. It is necessary for SUT to have access to the source code of the Magento instance. For example, you can install the SUT on one server and point it at your Magento installation on another server.

> 

Reviewer: If you are running SUT against a Magento customized instance with large third-party modules and files, the tool might require a high amount of RAM, at least 2GB RAM.

### Recommended actions

> It is recommended to avoid naming SUT modules with the same name, as a segmentation fault error can occur.

Reviewer: This is not a SUT issue it is a Magento limitation, the text should be changed for:
"Magento best practice is not to have 2 modules with the same name, if this happens SUT will show a segmentation fault error in which case you'll have to analyze each module independently with the option -m:"

@svera @jcuerdo @paliarush please validate

> If this segmentation error takes place, solve it by:

Reviewer: Please remove this:
"If this segmentation error takes place, solve it by:

*  Avoid analyzing a set of modules if there are 2 modules with the same name.
*  Analyze modules separately with the option -m:"

*  Avoid analyzing a set of modules if there are 2 modules with the same name.
*  Analyze modules separately with the option -m:

    ```bash
    bin/sut upgrade:check /(instance_path) --coming-version=2.4.1 -m /(module_path)
    ```

> 

Reviewer: Please add:
"If you get memory issues while executing SUT our recommendation is to use the -m command to run the tool against a specific module, this will prevent the problem to happen"

## Install

To install the SUT, you must install the necessary prerequisites:

*  Magento access keys
*  Composer
*  Node.js (only required to check GraphQL compatibility)

Refer to the [SUT installation]({{site.baseurl}}/safe-upgrade-tool/install.html#install).

### Magento access keys

> You must have [Magento access keys]({{site.baseurl}}/marketplace/sellers/profile-information.html#access-keys) to use the SUT. Add your Magento access keys to your `auth.json` file, which is located at `~/.composer` by default.

Reviewer: let's add download before use:
"to download and use the SUT."

---
group: software-update-guide
title: Install
functional_areas:
  - Upgrade
---

> The Safe Upgrade Tool (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed on it.

Reviewer: Could we add "(ALPHA)" next to Safe Upgrade Tool?

## Workflow

The following diagram shows the expected workflow when running the SUT:

![SUT Diagram](img/mvp-diagram.png)

### Who is the SUT for?

> The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:

Reviewer 2: The described scenario is our "desired state", not a reality with current version of SUT.

At the moment SUT is capable of analyzing only a fraction of all extensibility scenarios. It means that system integrators must not exclusively rely on SUT output to forecast timeline and cost of the upgrade.

This may be important, depending on our goal with this documentation.

Author: I would understand that being an ALPHA, that section would be ideal scenario, which we can still keep, if @sgonzalezmangana thinks worth it, we could add a clarification for it

> 1. A partner's software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.

Reviewer 3: ```suggestion
1. A partner's Software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.
```

> 1. The Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.

Reviewer 3: ```suggestion
1. The Software Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.
```

> 1. With this information, the Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.

Reviewer 3: ```suggestion
1. With this information, the Software Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.
```

1. The Account Manager creates a timeline and cost for the Magento upgrade, which allows them to get their manager's approval.

> 1. With their manager's approval, the Engineer works on the required code modifications to fix the broken modules.

Reviewer 3: ```suggestion
1. With their manager's approval, the Software Engineer works on the required code modifications to fix the broken modules.
```

> 1. The Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.

Reviewer 3: ```suggestion
1. The Software Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.
```

> 1. Everything check’s out and the Engineeer pushes the code to a staging environment where automated tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.

Reviewer 3: ```suggestion
1. Everything check’s out and the Software Engineer pushes the code to a staging environment where automated tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.
```

> 1. Everything check’s out and the Engineeer pushes the code to a staging environment where automated tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.

Reviewer 2: The reality is almost no system integrator has automated tests. Again, this is desired state. 

It would be more accurate to mention "regression testing" phase without specifying whether it is completed manually or automatically.

## Prerequisites

See [prerequisites]({{ site.baseurl }}/safe-upgrade-tool/prerequisites.html).

{:.bs-callout-info}

> You can run the SUT in any operating system. There is no requirement to run the SUT where your Magento instance is located. For example, you can install the SUT on one server and point it at your Magento installation on another server.

Reviewer 2: It is necessary for SUT to have access to the source code of the Magento instance.

## Install

To install the SUT, you must install the necessary prerequisites:

*  Magento access keys
*  Composer

> *  Node.js

Reviewer 2: Node JS is only a requirement if they wish to check GraphQL compatibility, otherwise it is not needed.

### Magento access keys

You must have [Magento access keys]({{ site.baseurl }}/marketplace/sellers/profile-information.html#access-keys) to use the SUT. Add your Magento access keys to your `auth.json` file, which is located at `~/.composer` by default.

{:.bs-callout-warning}
Check your **COMPOSER_HOME** environment variable to see where the `auth.json` file is located.

The **public key** corresponds to the _username_ whereas the **private key** is the _password_:

### Example of Magento access keys

```json
    "http-basic": {
        "repo.magento.com": {
            "username": "YOUR_MAGENTO_PUBLIC_KEY",
            "password": "YOUR_MAGENTO_PRIVATE_KEY"
        }
    },
```

### Composer

Clone the [safe-upgrade-tool](https://github.com/magento-commerce/safe-upgrade-tool) repository and run `composer install` in your terminal to install dependencies.

{:.bs-callout-warning}
If the **Magento access keys** are not correctly configured, the SUT will not install and you will get errors when running the `composer install` command.

### Node.js

To install Node.js, see the Node.js [documentation](https://nodejs.dev/learn/how-to-install-nodejs).

## Third-party extensions

> Magento recommends that you contact your search engine vendor to determine whether your extension is fully compatible with Magento 2.4.

Reviewer 2: Not sure why we have this section about "contacting search engine vendor", please clarify.

Author: As I understood it, contact your 3rd party vendor to check the extension is compatible with 2.4?.

Reviewer: We should remove the "Third-party extensions" content as it is not relevant at this point for SUT
